### PR TITLE
refactor: profile overview page

### PR DESCRIPTION
**Pull request by Max-vS**
image

### CODE REVIEW

File: app/app/me/[profile_id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"use client";
+
+import ProfileOverview from "../components/ProfileOverview";
+import { Profile } from "../../../models/profile";
+
+const Me = () => {
+  /*
```
Comment (softdevtumai) [inline app/app/me/[profile_id]/page.tsx]: Please remove commented out code if not used

File: app/app/me/components/ProfileEditor.tsx [hunk 2]
```diff
@@ -1,142 +1,50 @@
...
-      type: Yup.string().required("Type is required."),
-      link: Yup.string()
-        .url("Invalid URL format")
-        .required("Link is required."),
+import {UploadIcon, DownloadIcon} from "@radix-ui/react-icons";
+import { z } from 'zod';
+import { MeModel } from "@models/me";
+
+const profileSchema = z.object({
+  /*
```
Comment (softdevtumai) [inline app/app/me/components/ProfileEditor.tsx]: dead code

File: app/app/me/components/ProfileEditor.tsx [hunk 3]
```diff
@@ -1,142 +1,50 @@
...
+  profile_picture: z.union([z.string().nullable(), z.instanceof(FileList).optional()])
+    .refine((file) => !file || (file instanceof FileList && file[0].size <= 0.2 * 1048576), {
+      message: "File is too big! Max size is 200 KB.",
     }),
-  ),
+  */
+  first_name: z.string().nonempty("First name is required."),
```
Comment (softdevtumai) [inline app/app/me/components/ProfileEditor.tsx:15]: Whats the difference between `.nonempty` and `.required?`

Comment (softdevtumai) [reply app/app/me/components/ProfileEditor.tsx:15]: Thanks a lot for switching to zod with react-forms from Formik! Very clean

File: app/app/me/components/ProfileEditor.tsx [hunk 4]
```diff
@@ -1,142 +1,50 @@
...
-              values.profile_picture,
-            );
-          }
-          await meModel.updateEditorProfile(values);
-          await meModel.editProfile();
-          await queryClient.invalidateQueries({ queryKey: ["me"] });
-        }}
+        initialValues={profile}
+        validationSchema={profileSchema}
+        onSubmit={async (values) => {}}
```
Comment (softdevtumai) [inline app/app/me/components/ProfileEditor.tsx]: Maybe add something like this so we don't forget later ;)
```suggestion
        onSubmit={async (values) => {throw Error("TODO: Not implemented - update profile in DB")}}
```

**Review by softdevtumai — APPROVED** [review]
Amazing work. Looks awesome and code quality is top notch

File: app/app/me/components/ProfileEditor.tsx (hunk 4 above)
Comment (Max-vS) [reply app/app/me/components/ProfileEditor.tsx]: Good point, will do! Thanks

File: app/app/me/components/ProfileEditor.tsx (hunk 3 above)
Comment (Max-vS) [reply app/app/me/components/ProfileEditor.tsx:15]: nonempty prevents empty inputs, e.g. an empty string, which "required" doesn't necessarily do

**Pull request closed without merging**